### DISCUSSION TITLE
Simplify demo version downloading on Android

### DISCUSSION THREAD

**Discussion by ihhub** [ihhub/fheroes2#8534] [General]
Based on multiple users feedback the way how demo version of the game being downloaded is not intuitive and they are lost what exactly to do.

So, the main question is whether it is possible to modify our Android (Java) source code in order to be able to download and unpack demo version without the need of using browsers.

What do you think @oleg-derevenetz ?

**Comment by oleg-derevenetz** [comment]
One of the main reasons why I would like to avoid downloading something from the Internet inside our application is to minimize the set of permissions. In Android, permissions are set for the entire app, not on per-Activity basis, so if we give the application the `Manifest.permission.INTERNET` permission - for example, to use the [DownloadManager](https://developer.android.com/reference/android/app/DownloadManager) class, It will also be given to those parts of the app that don't need it. Extra permissions are always a bad thing, therefore, I preferred to use external apps (browsers/download managers/..., whatever the user prefers) specifically designed for the only task that requires this permission.

**Reply by ihhub** [reply]
We can delay adding INTERNET permission till we implement network part of the game.

**Reply by oleg-derevenetz** [reply]
I know that many people are paranoid about permissions (including me), and indeed, at the moment there is no real reason to grant fheroes2 the INTERNET permission from the point of view of the average user :) On Android, permission to access the Internet is considered "non-dangerous", i.e. it does not require additional confirmation from the user, it cannot be granted temporarily ("only this time"), etc.
